### ISSUE TITLE
 [ERROR] No matching export in "node_modules/react-native-screens/src/index.native.tsx" for import "isSearchBarAvailableForCurrentPlatform"      

### ISSUE THREAD

**Issue by ko-devHong** [oblador/react-native-esbuild#33]
I tried to run it after changing the`react-native-config.js` setting after a normal installation, but an error occurs during compilation.

<img width="991" alt="스크린샷 2023-09-20 오전 9 16 28" src="https://github.com/oblador/react-native-esbuild/assets/45538347/08dbb73f-1287-4af6-a8db-28a992975d7c">

- react-native.config.js
```js
const { createEsbuildCommands, defaultHasFlowSyntax, syntaxAwareLoaderPlugin } = require("react-native-esbuild");

const FLOW_MODULES_WITHOUT_PRAGMA = ["react-native-appsflyer"];

const commands = createEsbuildCommands((config, args) => ({
    ...config,
    plugins: config.plugins
        .filter(plugin => plugin.name !== "syntax-aware-loader")
        .concat(
            syntaxAwareLoaderPlugin({
                filter: /\.([mc]js|[tj]sx?)$/,
                cache: args.dev,
                hasFlowSyntax: (contents, filePath) =>
                    defaultHasFlowSyntax(contents, filePath) ||
                    FLOW_MODULES_WITHOUT_PRAGMA.find(m => filePath.includes(`node_modules/${m}/`))
            })
        )
}));

module.exports = {
    commands,
    project: {
        ios: {},
        android: {}
    },
    assets: ["./src/assets/fonts/"],
    dependencies: {
        ...(process.env.NO_FLIPPER ? { "react-native-flipper": { platforms: { ios: null } } } : {})
    }
};
```

- package.json
```json
{
 ...
 "react-native-screens": "^3.19.0",
 "@react-navigation/bottom-tabs": "^6.3.1",
 "@react-navigation/elements": "^1.3.13",
 "@react-navigation/material-top-tabs": "^6.2.2",
 "@react-navigation/native": "^6.1.2",
 "@react-navigation/native-stack": "^6.9.8",
 "react-native": "0.71.1",
}
```

My guess is that I don't support `esm`.

**Comment by ko-devHong** [comment]
I changed library code. 

![스크린샷 2023-09-20 오전 11 47 55](https://github.com/oblador/react-native-esbuild/assets/45538347/7d09ef5e-4508-4fa2-9463-d73cf7f35003)

But it is  not work

**Comment by ko-devHong** [comment]
@oblador  how can I solve this problem?  
I think it's a similar situation to the link issue below but I have too much to change the code in the library. 
 https://github.com/evanw/esbuild/issues/890

**Comment by LeoInCode** [comment]
I have the same experience. Idk more what I have to do. I think could be:
- When run "react-native esbuild-start", they cannot see tsconfig.json
- After running,  "react-native-.config.js" file don't execute metro, so, they cannot understand this modules

I visited many sites, looked all files of this library, and other projects who used this library, but no one could resolve.

@oblador , we need an answer about that!

**Comment by oblador** [comment]
To my understanding this is rooted in re-exporting types. I believe replacing them with explicit `export/import type` where needed solves it.

**Comment by oblador** [comment]
Regardless, if you want my help i need a minimal code example reproducing the error

**Comment by LeoInCode** [comment]
Yeah, I understand your explanation, but I cannot modify all my files of node_modules to adapt to this. Also, I discovered that this error is originally from esbuild. So, as you use that, you inherited this error.

**Comment by oblador** [comment]
node_modules in absolute majority of cases will ship transpiled code. I've worked in very large code bases and encountered this problem (which is not specific to just esbuild actually), but never in a third party lib. If it indeed is the case that they are _only_ shipping TS, it's _possible_ to patch it with `patch-package` or similar.

**Comment by LeoInCode** [comment]
But a patch-package for one by one of the erros, in the start of the project, is a shot in the foot.
